lib/reek/ast/sexp_extensions/methods.rb

```diff
@@ -30,8 +30,8 @@ def ends_with_bang?
         end
 
         def body_nodes(types, ignoring = [])
-          if body
-            body.find_nodes(types, ignoring)
+          if body && !ignoring.include?(body.type)
+            body.each_node(types, ignoring | types)
           else
             []
           end
```

Comment:
I’m not sure whether I wouldn’t prefer the more explicit handling of edge cases:
```ruby
def body_nodes(types, ignoring = [])
  return [] if ignoring.include?(body.type)
  return [] unless body
  body.each_node(types, ignoring | types)
end
```
(In a better world `body` would be non-nilable and an ‘empty’ body would return `[]` on `each_node` – at that point the second `return` line could be dropped altogether.)